engines/gob/dbase.cpp

```diff
@@ -27,8 +27,121 @@
 
 #include "gob/dbase.h"
 
+#include "common/tokenizer.h"
+
 namespace Gob {
 
+dbaseMultipeIndex::dbaseMultipeIndex() {
+	clear();
+}
+
+void dbaseMultipeIndex::clear() {
+	memset(&_creationDate, 0, sizeof(_creationDate));
+	memset(&_lastUpdate, 0, sizeof(_lastUpdate));
+
+	_version = 0;
+}
+
+// Supported key definition syntax:
+// key_definition ::= field_definition { "+" field_definition }
+// field_definition ::= field_name | STR(field_name, length, 0)
+Common::Array<dbaseMultipeIndex::FieldReference> dbaseMultipeIndex::parseKeyDefinition(const Common::String &keyDefinition) {
+	Common::Array<FieldReference> fieldReferences;
+	Common::StringTokenizer tokenizer(keyDefinition, "+");
+	while (!tokenizer.empty()) {
+		Common::String token = tokenizer.nextToken();
+		if (token.hasPrefix("STR(")) {
+			// STR(field_name, length, 0) expression
+			size_t firstCommaPos = token.find(',');
+			size_t secondCommaPos = token.find(',', firstCommaPos + 1);
+			Common::String fieldName = token.substr(4, firstCommaPos - 4);
+			size_t length = 0;
+			if (secondCommaPos != token.npos)
+				length = atoi(token.substr(firstCommaPos + 1, secondCommaPos - firstCommaPos - 1).c_str());
+
+			fieldReferences.push_back({fieldName, length});
+		} else {
+			// Field name only
+			fieldReferences.push_back({token, 0});
+		}
+	}
+
+	return fieldReferences;
+}
+
+const Common::Array<dbaseMultipeIndex::FieldReference>* dbaseMultipeIndex::getTagKeyDefinition(Common::String tagName) const
+{
+	if (!_tagKeyDefinitions.contains(tagName))
+		return nullptr;
+	else
+		return &_tagKeyDefinitions[tagName];
+}
+
+bool dbaseMultipeIndex::load(Common::SeekableReadStream &stream) {
+	_version = stream.readByte();
+	_creationDate.tm_year = stream.readByte();
+	_creationDate.tm_mon  = stream.readByte() - 1;
+	_creationDate.tm_mday = stream.readByte();
+
+	uint32 pos = stream.pos();
+	_dataFilename = stream.readString('\0', 16);
+	stream.seek(pos + 16);
+	stream.skip(2); // Block size
+	stream.skip(2); // Page size
+	stream.skip(1); // Production flag
+	stream.skip(1); // Max number of tags
+	stream.skip(1); // Tag length
+	stream.skip(1); // Reserved
+	_nbrOfTagsInUse = stream.readUint16LE(); // Number of tags in use
+	stream.skip(2); // Reserved
+	stream.skip(4); // Number of pages in tag file
+	stream.skip(4); // Pointer to first free page
+	stream.skip(4); // Number of available blocks
+
+	_lastUpdate.tm_year = stream.readByte();
+	_lastUpdate.tm_mon  = stream.readByte() - 1;
+	_lastUpdate.tm_mday = stream.readByte();
+
+	stream.skip(497); // Reserved
```

Comment:
So magic. Maybe it is equivalent to `stream.seek(512)`?

Reply:
Yes, stream.seek(512 + 32) actually.